Mathlib/SetTheory/Ordinal/CantorNormalForm.lean

```diff
@@ -57,6 +60,29 @@ theorem CNFRec_pos (b : Ordinal) {o : Ordinal} {C : Ordinal → Sort*} (ho : o 
     (H : ∀ o, o ≠ 0 → C (o % b ^ log b o) → C o) :
     @CNFRec b C H0 H o = H o ho (@CNFRec b C H0 H _) := by rw [CNFRec, dif_neg ho]
 
+/-- Inducts on the base `ω` expansion of an ordinal.
+
+This differs from `CNFRec` in that every instance of `ω ^ a` is considered separately. -/
```

Comment:
Is it not possible to consider eveery instance of `b ^ a` separately for a general base `b`?

Reply:
I've revamped this PR with said generalization.